pkg/validation/config.go

```diff
@@ -105,106 +134,125 @@ func validateBuildRootImageConfiguration(fieldRoot string, input *api.BuildRootI
 	}
 
 	if input.ProjectImageBuild != nil && input.ImageStreamTagReference != nil {
-		return []error{fmt.Errorf("%s: image_stream_tag and project_image are mutually exclusive", fieldRoot)}
-	}
-	if input.ProjectImageBuild != nil && input.FromRepository {
-		return []error{fmt.Errorf("%s: project_image and from_repository are mutually exclusive", fieldRoot)}
-	}
-	if input.FromRepository && input.ImageStreamTagReference != nil {
-		return []error{fmt.Errorf("%s: from_repository and image_stream_tag are mutually exclusive", fieldRoot)}
-	}
-	if input.ProjectImageBuild == nil && input.ImageStreamTagReference == nil && !input.FromRepository {
-		return []error{fmt.Errorf("%s: you have to specify one of project_image, image_stream_tag or from_repository", fieldRoot)}
-	}
-	if input.ImageStreamTagReference != nil {
-		return validateBuildRootImageStreamTag(fmt.Sprintf("%s.image_stream_tag", fieldRoot), *input.ImageStreamTagReference)
-	}
-	return nil
+		ret = append(ret, ctx.errorf("image_stream_tag and project_image are mutually exclusive"))
+	} else if input.ProjectImageBuild != nil && input.FromRepository {
+		ret = append(ret, ctx.errorf("project_image and from_repository are mutually exclusive"))
+	} else if input.FromRepository && input.ImageStreamTagReference != nil {
+		ret = append(ret, ctx.errorf("from_repository and image_stream_tag are mutually exclusive"))
+	} else if input.ProjectImageBuild == nil && input.ImageStreamTagReference == nil && !input.FromRepository {
+		ret = append(ret, ctx.errorf("you have to specify one of project_image, image_stream_tag or from_repository"))
+	} else if input.ImageStreamTagReference != nil {
+		ret = append(ret, validateBuildRootImageStreamTag(ctx.forField(".image_stream_tag"), *input.ImageStreamTagReference)...)
+	}
+	if err := ctx.addPipelineImage(api.PipelineImageStreamTagReferenceRoot); err != nil {
+		ret = append(ret, err)
+	}
+	return
 }
 
-func validateBuildRootImageStreamTag(fieldRoot string, buildRoot api.ImageStreamTagReference) []error {
+func validateBuildRootImageStreamTag(ctx *configContext, buildRoot api.ImageStreamTagReference) []error {
 	var validationErrors []error
 	if len(buildRoot.Namespace) == 0 {
-		validationErrors = append(validationErrors, fmt.Errorf("%s.namespace: value required but not provided", fieldRoot))
+		validationErrors = append(validationErrors, ctx.forField(".namespace").errorf("value required but not provided"))
```

Comment:
Why should the user have to worry about the `.` separator when adding context intead of `forField()` doing it for them?

Reply:
Created https://github.com/openshift/ci-tools/pull/2069 and modified the calls in this PR too.

All else, including tests, remain the same.